lib/logflare/endpoints/resolver.ex

```diff
@@ -4,35 +4,35 @@ defmodule Logflare.Endpoints.Resolver do
   """
   alias Logflare.Endpoints.Cache
 
-  def resolve(%Logflare.Endpoints.Query{id: id}) do
-    :global.registered_names()
-    |> Enum.filter(fn
-      {Cache, ^id, _} -> true
-      _ -> false
-    end)
-    |> Enum.map(&:global.whereis_name/1)
+  require Logger
+
+  @doc """
+  Lists all caches for an endpoint
+  """
+  def list_caches(%Logflare.Endpoints.Query{id: id}) do
+    :syn.members(:endpoints, id)
+    |> Enum.map(fn {pid, _} -> pid end)
   end
 
+  @doc """
+  Starts up or performs a lookup for an Endpoint.Cache process.
+  Returns the resolved pid.
+  """
   def resolve(%Logflare.Endpoints.Query{id: id} = query, params) do
-    :global.set_lock({Cache, {id, params}})
-
-    result =
-      case :global.whereis_name({Cache, id, params}) do
-        :undefined ->
-          spec = {Cache, {query, params}}
-
-          case DynamicSupervisor.start_child(Cache, spec) do
-            {:ok, pid} -> pid
-            {:error, {:already_started, pid}} -> pid
-          end
-
-        pid ->
-          Cache.touch(pid)
-          pid
-      end
+    :syn.lookup(:endpoints, {id, params})
```

Comment:
jfyi you [can attach a callback](https://hexdocs.pm/syn/syn_event_handler.html) to do something if two procs end up getting spun up.

We use this in Supavisor and Realtime for a custom resolver. Syn defaults to keeping the last process started (killing the first) and in both Supavisor and Realtime we want to keep the first. 

Just sharing for comms, in case you have to do something similar in the future. We might want to keep the first process but not necessary atm.